bee-api/bee-rest-api/src/endpoints/routes/api/v2/milestones_by_id.rs

```diff
@@ -24,7 +24,7 @@ async fn milestones_by_id<B: StorageBackend>(
 ) -> Result<impl IntoResponse, ApiError> {
     let milestone_payload = match args.tangle.get_milestone(milestone_id) {
```

Comment:
I think you can do `let milestone_payload =  args.tangle.get_milestone(milestone_id).ok_or_else(|| ApiError::NotFound)?;` instead.